src/dump.rs

```diff
@@ -207,6 +207,27 @@ impl DumpClassMember {
             },
         }
     }
+
+    pub fn add_tag(&mut self, tag: &str) {
+        let mut tags = match self {
+            DumpClassMember::Property(inner) => {
+                &mut inner.tags
+            },
+            DumpClassMember::Function(inner) => {
+                &mut inner.tags
+            },
+            DumpClassMember::Event(inner) => {
+                &mut inner.tags
+            },
+            DumpClassMember::Callback(inner) => {
+                &mut inner.tags
+            },
+        };
+        let tag: String = tag.to_owned();
+        if !tags.contains(&tag) {
+            tags.push(tag)
+        }

```

Comment:
We should be able to remove this type annotation, and probably move the `.to_owned()` call to be inside the `push` call.

Something like:

```rust
if !tags.contains(tags) {
    tags.push(tag.to_owned());
}
```

I also noticed that we're using `to_owned()` in quite a few places to check for membership in the set, which is allocating an extra string we shouldn't need. It's not so bad here since we're going to allocate that string anyways, but maybe we could do one of these as a bigger fix:

1. Switch `tags` to be a `HashSet` or `BTreeSet` (ordering would be nice!), which will give us quicker membership checks and helps us maintain our invariants of a tag only appearing once.
2. Change our membership checks to use an iterator like `tags.iter().any(|tag| tag == "Deprecated")`. This would probably be good to wrap up in a method like `has_tag(&self, tag: &str)` since it's a bit of a mouthful.

Reply:
Went with BTreeSet. Order probably doesn't matter until output, so might revisit that sometime later.